src/Ganeti/HTools/Loader.hs

```diff
@@ -388,11 +388,10 @@ setStaticKvmNodeMem nl static_node_mem =
 -- | Update node memory stat based on instance list.
 updateMemStat :: Node.Node -> Instance.List -> Node.Node
 updateMemStat node il =
-    let missing = truncate (Node.tMem node)
-                  - Node.nMem node
-                  - Node.fMem node
-                  - nodeImem node il
-    in node { Node.xMem = missing }
+  let node2 = node { Node.iMem = nodeImem node il }
```

Comment:
I was thinking about lenses because it would simplify each Node -> Node function you have here, so eventually you would be able to write `updateMemStat = fc . fb. fa`. However this is not as streamlined because of the dependency between each one of those. If you wanted to you would be able to use lambdas, but it would be as big as the code here. Feel free to ignore it.